remote/analyze_handler.py

```diff
@@ -94,42 +93,28 @@ def pre_analyze(analyze_id, part_number, workspace, use_cache):
 
     sources_root_path = os.path.join(analyze_dir_path, 'sources-root')
 
-    build_command_file_path = os.path.join(sources_root_path, 'build_command')
+    compile_commands_path = os.path.join(
+        sources_root_path, 'compile_commands.json')
 
-    with open(build_command_file_path) as build_action:
+    with open(compile_commands_path) as compile_commands_json:
         try:
-            build_command = build_action.readline()
+            compile_commands = json.loads(compile_commands_json.readline())[0]
         except Exception:
             REDIS_DATABASE.hset(analyze_id, 'state',
                                 AnalyzeStatus.PREANALYZE_FAILED.name)
-            LOG.error('Failed to read in build command.')
-            sys.exit(1)
-
-    file_to_analyze_path = os.path.join(sources_root_path, 'file_path')
-
-    with open(file_to_analyze_path) as read_in_file:
-        try:
-            read_in_file_path = read_in_file.readline()
-        except Exception:
-            REDIS_DATABASE.hset(analyze_id, 'state',
-                                AnalyzeStatus.PREANALYZE_FAILED.name)
-            LOG.error('Failed to read in file path.')
-            sys.exit(1)
+            LOG.error('Failed to read in compile command.')
```

Comment:
With handling an the index error separately you can log out easier if the compile command json was empty.